### DISCUSSION TITLE
Python async support

### DISCUSSION THREAD

**Discussion by ronenlinx** [arangodb/python-arango#345] [Q&A] [answered]
Hey,
I wanted to ask if there is any plan to support async operations in the near coming future?
If not, what is the best way to wrap the repo to use it asynchronously?

**Comment by apetenchea** [comment, marked as answer]
Hi @ronenlinx,

I'm sorry, I know this is probably late, I somehow missed your post. Will respond anyway in case other people stumble upon this discussion thread.

If you're interested in switching completely to asyncio, note that I released https://github.com/arangodb/python-arango-async and it is stable (that means there will be no drastic API changes). The documentation covers everything, and I made a "real-world" app to serve as a small tutorial: https://github.com/apetenchea/python-arango-async-demo

If you're interested to just wrap some bits of synchronous python-arango code in your application, I wrote about the recommended way to do that in this section of the documentation: https://python-arango-async.readthedocs.io/en/latest/migration.html#mixing-sync-and-async
